user-page/user_homepage/script.js

```diff
@@ -0,0 +1,175 @@
+    return dayName;
+};
+
+//get JS standard time format value
+function getJsTimeFormat(data) {
+    var monthName = getMonthName(data);
+    var day = getDayNumber(data);
+    var year = getYearNumber(data);
+    var x = new Date(monthName + " " + day + ", " + year);
+    return x;
+};
+
+//get year number based on date
+function getYearNumber(data) {
+    var year = "";
+    for (let i = 0; i < 4; i++) {
+        year = year + data.date[i];
+    };
+    return year;
+};
+
+//get month number based on date
+function getMonthNumber(data) {
+    var month = "";
+    for (let i = 5; i < 7; i++) {
+        month = month + data.date[i];
+    };
+    return month;
+
+};
+
+//get month name based on date
+function getMonthName(data) {
+    var monthList = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+    var month = ""; //set month as empty string
+
+    // go through data and extract numbers that corespond with month
+    //data dor month are on 5th and 6th place, therefore i starts at 5 and ends with 6
+    for (let i = 5; i < 7; i++) {
+        month = month + data.date[i];
+    };
+    month = month - 1; //substract one because first index of list is 0 not 1
+    month = monthList[month]; //find name of month that equals to the number
+    return month;
+};
+
+//get date of day (exapmle -> output = 15, only the number, not 'th or 'rd)
+function getDayNumber(data) {
+    var day = "";
+    for (let i = 8; i < 10; i++) {
+        day = day + data.date[i];
+    };
+    return day;
+};
+
+// get hour of arrival from data
+function getHourInNumber(data) {
+    var hour = "";
+    for (let i = 0; i < 2; i++) {
+        hour = hour + data.time_in[i];
+    };
+    return hour;
+}
+
+//get hour of departure from data
+function getHourOutNumber(data) {
+    var hour = "";
+    for (let i = 0; i < 2; i++) {
+        hour = hour + data.time_out[i];
+    };
+    return hour;
+}
+
+//get minute of arrival from data
+function getMinuteInNumber(data) {
+    var minute = "";
+    for (let i = 3; i < 5; i++) {
+        minute = minute + data.time_in[i];
+    };
+    return minute;
+};
+
+// get minute of departure from data
+function getMinuteOutNumber(data) {
+    var minute = "";
+    for (let i = 3; i < 5; i++) {
+        minute = minute + data.time_out[i];
+    };
+    return minute;
+};
+
+function durationOfStay(data) {
+    //                                                  minutes *1 to make sure its integer and not string
+    var minutesStayed = ((getHourOutNumber(data) * 60) + getMinuteOutNumber(data) * 1) - ((getHourInNumber(data) * 60) + getMinuteInNumber(data) * 1);
+    var hoursStayed = Math.floor(minutesStayed / 60);
+    minutesStayed = minutesStayed - hoursStayed * 60;
+    var duration = hoursStayed + ":" + minutesStayed;
+    return duration;
+
+
+};
+
+//add History Log
+function appendHistoryLog(data, divId) {
+    var newDiv = document.createElement("div");
```

Comment:
Are you planning to replace the DIV element by table?

Reply:
Not really. With designs i sketched. Divs make more sense. I may use grid in the future. Divs however, can be maintained.